moveit_planners/pilz_industrial_motion_planner/src/trajectory_generator_circ.cpp

```diff
@@ -125,7 +126,12 @@ void TrajectoryGeneratorCIRC::extractMotionPlanInfo(const planning_scene::Planni
       info.goal_joint_position[joint_item.joint_name] = joint_item.position;
     }
 
-    computeLinkFK(robot_model_, info.link_name, info.goal_joint_position, info.goal_pose);
+    if (!computeLinkFK(robot_state, info.link_name, info.goal_joint_position, info.goal_pose))
+    {
+      std::ostringstream os;
+      os << "Failed to compute forward kinematics for link: " << info.link_name << " of goal joints";
+      throw CircForwardForGoalIncalculable(os.str());
+    }
```

Comment:
Instead of duplicating this code in all calls to `computeLinkFK()`, why don't you throw directly from within this function?

Reply:
Done :+1: